### PR TITLE
Distribution Rate test introduced

### PR DESCRIPTION
**Pull request by gluk256**

### CODE REVIEW

File: test/libwhisper/bloomFilter.cpp [hunk 1]
```diff
@@ -244,4 +244,53 @@ BOOST_AUTO_TEST_CASE(bloomFilterRaw)
 	BOOST_REQUIRE(!f.contains(b00110111));
 }
 
-BOOST_AUTO_TEST_SUITE_END()
\ No newline at end of file
+enum { DistributionTestSize = 8 };
```
Comment (gavofyork) [inline test/libwhisper/bloomFilter.cpp]: `static const int DistributionTestSize = 8;` is preferred as it doesn't break GCC.

Comment (arkpar) [inline test/libwhisper/bloomFilter.cpp]: why not just use const int?

**Comment by gluk256** [comment]
fixed

File: test/libwhisper/bloomFilter.cpp [hunk 2]
```diff
@@ -244,4 +244,53 @@ BOOST_AUTO_TEST_CASE(bloomFilterRaw)
...
+				}
+}
+
+BOOST_AUTO_TEST_CASE(distributionRate)
+{
+	cnote << "Testing Bloom Filter Distribution Rate...";
+
+	unsigned const c_arrSize = DistributionTestSize * 8;
+	unsigned distribution[c_arrSize];
+	memset(distribution, 0, c_arrSize * sizeof(unsigned));
```
Comment (subtly) [inline test/libwhisper/bloomFilter.cpp]: Why not array of unsigned?  Default allocator will set values to 0 and (typically) ensures for loop won't go out of bounds.

**Comment by gavofyork** [comment]
agreed with alex's comment.

**Comment by gluk256** [comment]
I have fixed accordingly.
However, my tests show that std::array does not set values to 0. You still have to initialize them.

**Comment by subtly** [comment]
Ah, sorry. That was std::vector. Array does have fill() :)

**Pull request merged**